build/modules/gke/cluster.tf

```diff
@@ -28,136 +28,70 @@ data "google_client_config" "default" {}
-  location    = "${google_container_cluster.primary.location}"
-  project  = "${lookup(var.cluster, "project")}"
-  provider = "google-beta"
-  node_count = 1
-  node_config = {
-    preemptible  = true
-    machine_type = "n1-standard-4"
-
-    oauth_scopes = [
-      "https://www.googleapis.com/auth/devstorage.read_only",
-      "https://www.googleapis.com/auth/logging.write",
-      "https://www.googleapis.com/auth/monitoring",
-      "https://www.googleapis.com/auth/service.management.readonly",
-      "https://www.googleapis.com/auth/servicecontrol",
-      "https://www.googleapis.com/auth/trace.append",
-    ]
-    labels = {
-      "agones.dev/agones-system" = "true"
-    }
-    taint = {
-        key = "agones.dev/agones-system"
-        value = "true"
-        effect = "NO_EXECUTE"
-    }
-  }
-}
+    node_config {
+      machine_type = "${var.cluster["machineType"]}"
 
-resource "google_container_node_pool" "agones-metrics" {
-  name       = "agones-metrics"
-  cluster   = "${google_container_cluster.primary.name}"
-  location    = "${google_container_cluster.primary.location}"
-  project  = "${lookup(var.cluster, "project")}"
-  provider = "google-beta"
-  node_count = 1
-  node_config = {
-    preemptible  = true
-    machine_type = "n1-standard-4"
-
-    oauth_scopes = [
-      "https://www.googleapis.com/auth/devstorage.read_only",
-      "https://www.googleapis.com/auth/logging.write",
-      "https://www.googleapis.com/auth/monitoring",
-      "https://www.googleapis.com/auth/service.management.readonly",
-      "https://www.googleapis.com/auth/servicecontrol",
-      "https://www.googleapis.com/auth/trace.append",
-    ]
-    labels = {
-      "agones.dev/agones-metrics" = "true"
+      oauth_scopes = [
+        "https://www.googleapis.com/auth/devstorage.read_only",
+        "https://www.googleapis.com/auth/logging.write",
+        "https://www.googleapis.com/auth/monitoring",
+        "https://www.googleapis.com/auth/service.management.readonly",
+        "https://www.googleapis.com/auth/servicecontrol",
+        "https://www.googleapis.com/auth/trace.append",
+      ]
+
+      tags = ["game-server"]
     }
-    taint = {
-        key = "agones.dev/agones-metrics"
-        value = "true"
+  }
+  node_pool {
+    name       = "agones-system"
+    node_count = 1
+
+    node_config {
+      machine_type = "n1-standard-4"
+
+      oauth_scopes = [
+        "https://www.googleapis.com/auth/devstorage.read_only",
+        "https://www.googleapis.com/auth/logging.write",
+        "https://www.googleapis.com/auth/monitoring",
+        "https://www.googleapis.com/auth/service.management.readonly",
+        "https://www.googleapis.com/auth/servicecontrol",
+        "https://www.googleapis.com/auth/trace.append",
+      ]
+
+      labels = {
+        "stable.agones.dev/agones-system" = "true"
+      }
+
+      taint {
+        key    = "stable.agones.dev/agones-system"
+        value  = "true"
         effect = "NO_EXECUTE"
+      }
```

Comment:
This seems to have lost the node pool for agones-metrics.

Reply:
that's right

Reply:
Was that on purpose or should the 3rd node pool be defined here?

Reply:
It should be there